### PR TITLE
feat: rewrite validate_proxy_domains celery task as temporal workflow

### PR DESCRIPTION
**Pull request by frankh**
## Problem

Towards https://github.com/PostHog/product-internal/issues/602

bonus: now using gRPC instead of http

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/api/proxy_record.py [hunk 1]
```diff
@@ -45,14 +50,34 @@ def list(self, request, *args, **kwargs):
...
+                inputs,
+                id=workflow_id,
+                task_queue=BATCH_EXPORTS_TASK_QUEUE,
+            )
+        )
+
+        return Response(
+            {"success": True},
+            status=status.HTTP_200_OK,
+        )
```
Comment (daibhin) [inline posthog/api/proxy_record.py:80]: Does this break the frontend? Looks like we're setting the `proxyRecords` to the result of this endpoint which could cause an issue.

Comment (frankh) [reply posthog/api/proxy_record.py:80]: ah i was wondering if it was intentional or not, i'll switch that back

Comment (frankh) [reply posthog/api/proxy_record.py:80]: looks like it's ignoring the result and re-fetching actually:

```
createRecordSuccess: () => actions.loadRecords(),
```

Comment (daibhin) [reply posthog/api/proxy_record.py:80]: Feel free to leave it if you like. I was going to pick off the frontend tasks (e.g. pay gate) so maybe it's best if you make all your changes and I can double back to finish off

**Review by daibhin — APPROVED** [review]
Not a temporal expert and have no real way of testing / running it but everything looks reasonable

File: posthog/temporal/proxy_service/delete.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    record matching the expected value.
+    """
+    logger = await bind_temporal_org_worker_logger(organization_id=inputs.organization_id)
+    logger.info(
+        "Deleting proxy record %s",
+        inputs.proxy_record_id,
+    )
+
+    pr = await sync_to_async(ProxyRecord.objects.get)(id=inputs.proxy_record_id)
+    await sync_to_async(pr.delete())
```
Comment (fuziontech) [inline posthog/temporal/proxy_service/delete.py:51]:
```suggestion
    await sync_to_async(pr.delete)()
```

**Review by fuziontech — APPROVED** [review]
This is some of the most easily readable code I've read here.
Also perfect application of Temporal and grpc 🙏

File: posthog/temporal/proxy_service/delete.py (hunk 2 above)
Comment (frankh) [reply posthog/temporal/proxy_service/delete.py]: ooooh that's what the time error was from 🤦

Comment (frankh) [reply posthog/temporal/proxy_service/delete.py]: well, my change fixed it anyway and i think it's more readable

**Pull request merged**